syscalls.c

```diff
@@ -173,52 +175,119 @@ int syscalls_release(void *ustack)
 }
 
 
+static int syscalls_sanitizeVector(process_t *proc, char **v)
```

Comment:
On top of vulnerabilities of `vm_mapStringBelongs` and `vm_mapStringBelongs`, this function is also vulnerable to user changing the elements of vector. Unfortunatelly, I think that there's no way around copying the vector, but a FIXME note should be sufficient for now.